meson: add page (#4281)
tldr-pages/tldr: pages/common/meson.md

# meson

> > SCons-like build system that use python as a front-end language and Ninja as a building backend.

Reviewer: Perhaps this would be more consistent with other existing pages?
```suggestion
> SCons-like build system that uses python as a front-end language and Ninja as a building backend.
```

> > More information: <https://mesonbuild.com/>.

Reviewer 2: ```suggestion
> More information: <https://mesonbuild.com>.
```

> - Generate a c project with name myproject and version 0.1:

Reviewer: Perhaps this will read better?
```suggestion
- Generate a c project with a given name and version:
```

`meson init --language={{c}} --name={{myproject}} --version={{0.1}}`

- Configure builddir with default values:

> `meson setup {{builddir}}`

Reviewer 2: I have a question: Should the user run this command in the working directory where the `meson.build` file is located and `builddir` is the convention recommended name?

Author: the user  can use `-C` specify directory where `meson.build` located, question 2 is yes.

Reviewer 2: because, based on our [style-guide](https://github.com/tldr-pages/tldr/blob/master/contributing-guides/style-guide.md), we recommend the use of `{{path/to/builddir}}` but there are exceptions to the rule and maybe this the case. What do you think @sbrl?

Author: @navarroaxel  sorry, I have made some mistatakes in previous reply. `meson.build` must locate in working directory ,`-C` is to specify build directory

Reviewer: Hrm. I think `{{path/to/build_dir}}` might be better.
```suggestion
`meson setup {{build_dir}}`
```

- Build the project:

> `meson compile -C {{builddir}}`

Reviewer: Same here :-)
```suggestion
`meson compile -C {{path/to/build_dir}}`
```

> - Show help message:

Reviewer: Perhaps this will read better?
```suggestion
- Show the help:
```

> - Show help message:

Reviewer: Perhaps this will read better?
```suggestion
- Show the help:
```
